src/components/FeatureViews/Registration/CourseList.js

```diff
@@ -177,25 +278,47 @@ const CourseList = ({ filteredCourses, updatedParent }) => {
                                 {course.enrollmentSet.length} / {course.maxCapacity}
                             </ListStatus>
                             <ListButton>
-                                <ResponsiveButton
-                                    disabled={shouldDisableQuickRegister({
-                                        course, enrolledCourseIds,
-                                        registrations, studentIdList
-                                    })}                                    
-                                    variant="contained"
-                                    onClick={handleStartQuickRegister(course.id)}
-                                    data-cy="quick-register-class"
-                                    startIcon={<AddIcon />}
-                                >
-                                    register
-                                </ResponsiveButton>
+                                {course.enrollmentSet.length < course.maxCapacity || !parentIsLoggedIn
+                                    ? <ResponsiveButton
+                                        disabled={shouldDisableQuickRegister({
+                                            course, enrolledCourseIds,
+                                            registrations, studentIdList
+                                        })}
+                                        variant="contained"
+                                        onClick={handleStartQuickRegister(course.id)}
+                                        data-cy="quick-register-class"
+                                        startIcon={<AddIcon />}
+                                    >
+                                        register
+                                    </ResponsiveButton>
+                                    : inParentInterestList(course.id)
+                                        ? <ResponsiveButton
```

Comment:
Avoid using a ternary in a ternary. create a small component called `<ParentCourseInterestBtn/>` that takes in the courseId and the onClick handlers to determine this logic so from 281 to 314 should just be switching between a Register and Interest button